### PR TITLE
Add the ability to access and filter courses by UUID and not just key.

### PR DESCRIPTION
**Pull request by mjfrey**
Add to the course API endpoint the ability to access a course directly via the UUID.
Also added a uuid course filter that adds the ability to provide a list of UUIDs and get back the information for the courses.

LEARNER-2882

### CODE REVIEW

File: course_discovery/apps/api/v1/views/courses.py [hunk 1]
```diff
@@ -16,15 +19,37 @@ class CourseViewSet(viewsets.ReadOnlyModelViewSet):
...
+
+        if self.course_key_regex.match(key):
+            filter_key = 'key'
+        elif self.course_uuid_regex.match(key):
+            filter_key = 'uuid'
+
+        filter_kwargs = {filter_key: key}
+        obj = get_object_or_404(queryset, **filter_kwargs)
+
+        # May raise a permission denied
```
Comment (staubina) [inline course_discovery/apps/api/v1/views/courses.py:47]: Should we be concerned with this raising the permission denied exception?

File: course_discovery/apps/api/v1/views/courses.py [hunk 2]
```diff
@@ -16,15 +19,37 @@ class CourseViewSet(viewsets.ReadOnlyModelViewSet):
     """ Course resource. """
     filter_backends = (DjangoFilterBackend,)
     filter_class = filters.CourseFilter
+    # We should eventually transition over to UUID as the PK.
```
Comment (staubina) [inline course_discovery/apps/api/v1/views/courses.py]: Is this comment of value in here? I find that these get lost and the value is diminishing.  Should we create a backlog Story or Epic instead?

**Review by staubina — COMMENTED** [review]
Some small questions and comments.  The changes seem straight forward.

File: course_discovery/apps/api/v1/views/courses.py (hunk 1 above)
Comment (mjfrey) [reply course_discovery/apps/api/v1/views/courses.py:47]: no, this is what the base DRF method does.

**Review by bderusha — APPROVED** [review]

**Review by staubina — APPROVED** [review]

**Pull request merged**